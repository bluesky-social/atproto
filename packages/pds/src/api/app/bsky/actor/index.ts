import { Server } from '../../../../lexicon'
import AppContext from '../../../../context'

import getPreferences from './getPreferences'
import getProfile from './getProfile'
import getProfiles from './getProfiles'
import getSuggestions from './getSuggestions'
import putPreferences from './putPreferences'
import searchActors from './searchActors'
import searchActorsTypeahead from './searchActorsTypeahead'

export default function (server: Server, ctx: AppContext) {
  getPreferences(server, ctx)
  getProfile(server, ctx)
  getProfiles(server, ctx)
  getSuggestions(server, ctx)
  putPreferences(server, ctx)
  searchActors(server, ctx)
  searchActorsTypeahead(server, ctx)
}
