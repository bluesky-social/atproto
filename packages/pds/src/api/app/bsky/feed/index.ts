import { Server } from '../../../../lexicon'
import AppContext from '../../../../context'
import getActorFeeds from './getActorFeeds'
import getActorLikes from './getActorLikes'
import getAuthorFeed from './getAuthorFeed'
import getFeed from './getFeed'
import getFeedGenerator from './getFeedGenerator'
import getFeedGenerators from './getFeedGenerators'
import getLikes from './getLikes'
import getListFeed from './getListFeed'
import getPosts from './getPosts'
import getPostThread from './getPostThread'
import getRepostedBy from './getRepostedBy'
import getSuggestedFeeds from './getSuggestedFeeds'
import getTimeline from './getTimeline'
import searchPosts from './searchPosts'

export default function (server: Server, ctx: AppContext) {
  getActorFeeds(server, ctx)
  getActorLikes(server, ctx)
  getAuthorFeed(server, ctx)
  getFeed(server, ctx)
  getFeedGenerator(server, ctx)
  getFeedGenerators(server, ctx)
  getLikes(server, ctx)
  getListFeed(server, ctx)
  getPosts(server, ctx)
  getPostThread(server, ctx)
  getRepostedBy(server, ctx)
  getSuggestedFeeds(server, ctx)
  getTimeline(server, ctx)
  searchPosts(server, ctx)
}
