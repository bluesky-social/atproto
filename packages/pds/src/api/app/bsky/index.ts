import { Server } from '../../../lexicon'
import AppContext from '../../../context'
import graph from './graph'
import notification from './notification'

export default function (server: Server, ctx: AppContext) {
  graph(server, ctx)
  notification(server, ctx)
}
