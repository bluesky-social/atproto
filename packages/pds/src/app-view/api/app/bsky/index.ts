import { Server } from '../../../../lexicon'
import AppContext from '../../../../context'
import getTimeline from './feed/getTimeline'
import getAuthorFeed from './feed/getAuthorFeed'
import getLikes from './feed/getLikes'
import getPostThread from './feed/getPostThread'
import getPosts from './feed/getPosts'
import getProfile from './actor/getProfile'
import getProfiles from './actor/getProfiles'
import getRepostedBy from './feed/getRepostedBy'
import getFollowers from './graph/getFollowers'
import getFollows from './graph/getFollows'
import getUsersSearch from './actor/searchActors'
import getUsersTypeahead from './actor/searchActorsTypeahead'
import getSuggestions from './actor/getSuggestions'
import unspecced from './unspecced'

export default function (server: Server, ctx: AppContext) {
  getTimeline(server, ctx)
  getAuthorFeed(server, ctx)
  getLikes(server, ctx)
  getPostThread(server, ctx)
  getPosts(server, ctx)
  getProfile(server, ctx)
  getProfiles(server, ctx)
  getRepostedBy(server, ctx)
  getFollowers(server, ctx)
  getFollows(server, ctx)
  getUsersSearch(server, ctx)
  getUsersTypeahead(server, ctx)
  getSuggestions(server, ctx)
  unspecced(server, ctx)
}
