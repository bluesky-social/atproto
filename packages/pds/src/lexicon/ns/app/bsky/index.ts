/**
 * GENERATED CODE - DO NOT MODIFY
 */
import {
  type Auth,
  type Options as XrpcOptions,
  Server as XrpcServer,
  type StreamConfigOrHandler,
  type MethodConfigOrHandler,
  createServer as createXrpcServer,
} from '@atproto/xrpc-server'
import { Server } from '../../../index.js'
import { AppBskyActorNS } from './actor/index.js'
import { AppBskyEmbedNS } from './embed/index.js'
import { AppBskyFeedNS } from './feed/index.js'
import { AppBskyGraphNS } from './graph/index.js'
import { AppBskyLabelerNS } from './labeler/index.js'
import { AppBskyNotificationNS } from './notification/index.js'
import { AppBskyRichtextNS } from './richtext/index.js'
import { AppBskyUnspeccedNS } from './unspecced/index.js'
import { AppBskyVideoNS } from './video/index.js'

export class AppBskyNS {
  _server: Server
  actor: AppBskyActorNS
  embed: AppBskyEmbedNS
  feed: AppBskyFeedNS
  graph: AppBskyGraphNS
  labeler: AppBskyLabelerNS
  notification: AppBskyNotificationNS
  richtext: AppBskyRichtextNS
  unspecced: AppBskyUnspeccedNS
  video: AppBskyVideoNS

  constructor(server: Server) {
    this._server = server
    this.actor = new AppBskyActorNS(server)
    this.embed = new AppBskyEmbedNS(server)
    this.feed = new AppBskyFeedNS(server)
    this.graph = new AppBskyGraphNS(server)
    this.labeler = new AppBskyLabelerNS(server)
    this.notification = new AppBskyNotificationNS(server)
    this.richtext = new AppBskyRichtextNS(server)
    this.unspecced = new AppBskyUnspeccedNS(server)
    this.video = new AppBskyVideoNS(server)
  }
}

export * from './actor/index.js'
export * from './embed/index.js'
export * from './feed/index.js'
export * from './graph/index.js'
export * from './labeler/index.js'
export * from './notification/index.js'
export * from './richtext/index.js'
export * from './unspecced/index.js'
export * from './video/index.js'
