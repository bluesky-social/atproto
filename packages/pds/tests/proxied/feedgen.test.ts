import { makeAlgos } from '@atproto/bsky'
import AtpAgent, { AtUri, FeedNS } from '@atproto/api'
import { TestNetwork, SeedClient } from '@atproto/dev-env'
import basicSeed from '../seeds/basic'
import { forSnapshot } from '../_util'

describe('feedgen proxy view', () => {
  let network: TestNetwork
  let agent: AtpAgent
  let sc: SeedClient

  const origGetFeedGenerator = FeedNS.prototype.getFeedGenerator
  const feedUri = AtUri.make(
    'did:example:feed-publisher',
    'app.bsky.feed.generator',
    'mutuals',
  )

  beforeAll(async () => {
    network = await TestNetwork.create({
      dbPostgresSchema: 'proxy_feedgen',
      bsky: { algos: makeAlgos(feedUri.host) },
    })
    agent = network.pds.getClient()
    sc = network.getSeedClient()
    await basicSeed(sc, { addModLabels: true })
    // publish feed
    const feed = await agent.api.app.bsky.feed.generator.create(
      { repo: sc.dids.alice, rkey: feedUri.rkey },
      {
        did: network.bsky.ctx.cfg.feedGenDid ?? '',
        displayName: 'Mutuals',
        createdAt: new Date().toISOString(),
      },
      sc.getHeaders(sc.dids.alice),
    )
    await network.processAll()
    // mock getFeedGenerator() for use by pds's getFeed since we don't have a proper feedGenDid or feed publisher
    FeedNS.prototype.getFeedGenerator = async function (params, opts) {
      if (params?.feed === feedUri.toString()) {
        return {
          success: true,
          data: {
            isOnline: true,
            isValid: true,
            view: {
              cid: feed.cid,
              uri: feed.uri,
              did: network.bsky.ctx.cfg.feedGenDid ?? '',
              creator: { did: sc.dids.alice, handle: 'alice.test' },
              displayName: 'Mutuals',
              indexedAt: new Date().toISOString(),
            },
          },
          headers: {},
        }
      }
      return origGetFeedGenerator.call(this, params, opts)
    }
  })

  afterAll(async () => {
    FeedNS.prototype.getFeedGenerator = origGetFeedGenerator
    await network.close()
  })

  it('performs basic proxy of getFeed', async () => {
    const { data: feed } = await agent.api.app.bsky.feed.getFeed(
      { feed: feedUri.toString() },
      {
        headers: { ...sc.getHeaders(sc.dids.alice) },
      },
    )
    expect(forSnapshot(feed)).toMatchSnapshot()
  })
})
