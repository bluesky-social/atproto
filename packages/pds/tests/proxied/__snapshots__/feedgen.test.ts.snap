// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feedgen proxy view performs basic proxy of getFeed 1`] = `
Object {
  "cursor": "0000000000000::bafycid",
  "feed": Array [
    Object {
      "post": Object {
        "author": Object {
          "avatar": "https://bsky.public.url/img/avatar/plain/user(1)/cids(1)@jpeg",
          "did": "user(0)",
          "displayName": "ali",
          "handle": "alice.test",
          "labels": Array [
            Object {
              "cid": "cids(2)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "user(0)",
              "uri": "record(1)",
              "val": "self-label-a",
            },
            Object {
              "cid": "cids(2)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "user(0)",
              "uri": "record(1)",
              "val": "self-label-b",
            },
          ],
          "viewer": Object {
            "blockedBy": false,
            "muted": false,
          },
        },
        "cid": "cids(0)",
        "indexedAt": "1970-01-01T00:00:00.000Z",
        "labels": Array [],
        "likeCount": 0,
        "record": Object {
          "$type": "app.bsky.feed.post",
          "createdAt": "1970-01-01T00:00:00.000Z",
          "reply": Object {
            "parent": Object {
              "cid": "cids(4)",
              "uri": "record(3)",
            },
            "root": Object {
              "cid": "cids(3)",
              "uri": "record(2)",
            },
          },
          "text": "thanks bob",
        },
        "replyCount": 0,
        "repostCount": 1,
        "uri": "record(0)",
        "viewer": Object {},
      },
      "reply": Object {
        "parent": Object {
          "$type": "app.bsky.feed.defs#postView",
          "author": Object {
            "avatar": "https://bsky.public.url/img/avatar/plain/user(3)/cids(1)@jpeg",
            "did": "user(2)",
            "displayName": "bobby",
            "handle": "bob.test",
            "labels": Array [],
            "viewer": Object {
              "blockedBy": false,
              "followedBy": "record(5)",
              "following": "record(4)",
              "muted": false,
            },
          },
          "cid": "cids(4)",
          "embed": Object {
            "$type": "app.bsky.embed.images#view",
            "images": Array [
              Object {
                "alt": "tests/sample-img/key-landscape-small.jpg",
                "fullsize": "https://bsky.public.url/img/feed_fullsize/plain/user(3)/cids(5)@jpeg",
                "thumb": "https://bsky.public.url/img/feed_thumbnail/plain/user(3)/cids(5)@jpeg",
              },
            ],
          },
          "indexedAt": "1970-01-01T00:00:00.000Z",
          "labels": Array [
            Object {
              "cid": "cids(4)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "did:example:labeler",
              "uri": "record(3)",
              "val": "test-label",
            },
            Object {
              "cid": "cids(4)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "did:example:labeler",
              "uri": "record(3)",
              "val": "test-label-2",
            },
          ],
          "likeCount": 0,
          "record": Object {
            "$type": "app.bsky.feed.post",
            "createdAt": "1970-01-01T00:00:00.000Z",
            "embed": Object {
              "$type": "app.bsky.embed.images",
              "images": Array [
                Object {
                  "alt": "tests/sample-img/key-landscape-small.jpg",
                  "image": Object {
                    "$type": "blob",
                    "mimeType": "image/jpeg",
                    "ref": Object {
                      "$link": "cids(5)",
                    },
                    "size": 4114,
                  },
                },
              ],
            },
            "reply": Object {
              "parent": Object {
                "cid": "cids(3)",
                "uri": "record(2)",
              },
              "root": Object {
                "cid": "cids(3)",
                "uri": "record(2)",
              },
            },
            "text": "hear that label_me label_me_2",
          },
          "replyCount": 1,
          "repostCount": 0,
          "uri": "record(3)",
          "viewer": Object {},
        },
        "root": Object {
          "$type": "app.bsky.feed.defs#postView",
          "author": Object {
            "avatar": "https://bsky.public.url/img/avatar/plain/user(1)/cids(1)@jpeg",
            "did": "user(0)",
            "displayName": "ali",
            "handle": "alice.test",
            "labels": Array [
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-a",
              },
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-b",
              },
            ],
            "viewer": Object {
              "blockedBy": false,
              "muted": false,
            },
          },
          "cid": "cids(3)",
          "indexedAt": "1970-01-01T00:00:00.000Z",
          "labels": Array [],
          "likeCount": 3,
          "record": Object {
            "$type": "app.bsky.feed.post",
            "createdAt": "1970-01-01T00:00:00.000000Z",
            "text": "again",
          },
          "replyCount": 2,
          "repostCount": 1,
          "uri": "record(2)",
          "viewer": Object {},
        },
      },
    },
    Object {
      "post": Object {
        "author": Object {
          "did": "user(4)",
          "handle": "carol.test",
          "labels": Array [],
          "viewer": Object {
            "blockedBy": false,
            "followedBy": "record(8)",
            "following": "record(7)",
            "muted": false,
          },
        },
        "cid": "cids(6)",
        "indexedAt": "1970-01-01T00:00:00.000Z",
        "labels": Array [],
        "likeCount": 0,
        "record": Object {
          "$type": "app.bsky.feed.post",
          "createdAt": "1970-01-01T00:00:00.000Z",
          "reply": Object {
            "parent": Object {
              "cid": "cids(3)",
              "uri": "record(2)",
            },
            "root": Object {
              "cid": "cids(3)",
              "uri": "record(2)",
            },
          },
          "text": "of course",
        },
        "replyCount": 0,
        "repostCount": 0,
        "uri": "record(6)",
        "viewer": Object {},
      },
      "reply": Object {
        "parent": Object {
          "$type": "app.bsky.feed.defs#postView",
          "author": Object {
            "avatar": "https://bsky.public.url/img/avatar/plain/user(1)/cids(1)@jpeg",
            "did": "user(0)",
            "displayName": "ali",
            "handle": "alice.test",
            "labels": Array [
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-a",
              },
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-b",
              },
            ],
            "viewer": Object {
              "blockedBy": false,
              "muted": false,
            },
          },
          "cid": "cids(3)",
          "indexedAt": "1970-01-01T00:00:00.000Z",
          "labels": Array [],
          "likeCount": 3,
          "record": Object {
            "$type": "app.bsky.feed.post",
            "createdAt": "1970-01-01T00:00:00.000000Z",
            "text": "again",
          },
          "replyCount": 2,
          "repostCount": 1,
          "uri": "record(2)",
          "viewer": Object {},
        },
        "root": Object {
          "$type": "app.bsky.feed.defs#postView",
          "author": Object {
            "avatar": "https://bsky.public.url/img/avatar/plain/user(1)/cids(1)@jpeg",
            "did": "user(0)",
            "displayName": "ali",
            "handle": "alice.test",
            "labels": Array [
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-a",
              },
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-b",
              },
            ],
            "viewer": Object {
              "blockedBy": false,
              "muted": false,
            },
          },
          "cid": "cids(3)",
          "indexedAt": "1970-01-01T00:00:00.000Z",
          "labels": Array [],
          "likeCount": 3,
          "record": Object {
            "$type": "app.bsky.feed.post",
            "createdAt": "1970-01-01T00:00:00.000000Z",
            "text": "again",
          },
          "replyCount": 2,
          "repostCount": 1,
          "uri": "record(2)",
          "viewer": Object {},
        },
      },
    },
    Object {
      "post": Object {
        "author": Object {
          "avatar": "https://bsky.public.url/img/avatar/plain/user(3)/cids(1)@jpeg",
          "did": "user(2)",
          "displayName": "bobby",
          "handle": "bob.test",
          "labels": Array [],
          "viewer": Object {
            "blockedBy": false,
            "followedBy": "record(5)",
            "following": "record(4)",
            "muted": false,
          },
        },
        "cid": "cids(4)",
        "embed": Object {
          "$type": "app.bsky.embed.images#view",
          "images": Array [
            Object {
              "alt": "tests/sample-img/key-landscape-small.jpg",
              "fullsize": "https://bsky.public.url/img/feed_fullsize/plain/user(3)/cids(5)@jpeg",
              "thumb": "https://bsky.public.url/img/feed_thumbnail/plain/user(3)/cids(5)@jpeg",
            },
          ],
        },
        "indexedAt": "1970-01-01T00:00:00.000Z",
        "labels": Array [
          Object {
            "cid": "cids(4)",
            "cts": "1970-01-01T00:00:00.000Z",
            "neg": false,
            "src": "did:example:labeler",
            "uri": "record(3)",
            "val": "test-label",
          },
          Object {
            "cid": "cids(4)",
            "cts": "1970-01-01T00:00:00.000Z",
            "neg": false,
            "src": "did:example:labeler",
            "uri": "record(3)",
            "val": "test-label-2",
          },
        ],
        "likeCount": 0,
        "record": Object {
          "$type": "app.bsky.feed.post",
          "createdAt": "1970-01-01T00:00:00.000Z",
          "embed": Object {
            "$type": "app.bsky.embed.images",
            "images": Array [
              Object {
                "alt": "tests/sample-img/key-landscape-small.jpg",
                "image": Object {
                  "$type": "blob",
                  "mimeType": "image/jpeg",
                  "ref": Object {
                    "$link": "cids(5)",
                  },
                  "size": 4114,
                },
              },
            ],
          },
          "reply": Object {
            "parent": Object {
              "cid": "cids(3)",
              "uri": "record(2)",
            },
            "root": Object {
              "cid": "cids(3)",
              "uri": "record(2)",
            },
          },
          "text": "hear that label_me label_me_2",
        },
        "replyCount": 1,
        "repostCount": 0,
        "uri": "record(3)",
        "viewer": Object {},
      },
      "reply": Object {
        "parent": Object {
          "$type": "app.bsky.feed.defs#postView",
          "author": Object {
            "avatar": "https://bsky.public.url/img/avatar/plain/user(1)/cids(1)@jpeg",
            "did": "user(0)",
            "displayName": "ali",
            "handle": "alice.test",
            "labels": Array [
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-a",
              },
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-b",
              },
            ],
            "viewer": Object {
              "blockedBy": false,
              "muted": false,
            },
          },
          "cid": "cids(3)",
          "indexedAt": "1970-01-01T00:00:00.000Z",
          "labels": Array [],
          "likeCount": 3,
          "record": Object {
            "$type": "app.bsky.feed.post",
            "createdAt": "1970-01-01T00:00:00.000000Z",
            "text": "again",
          },
          "replyCount": 2,
          "repostCount": 1,
          "uri": "record(2)",
          "viewer": Object {},
        },
        "root": Object {
          "$type": "app.bsky.feed.defs#postView",
          "author": Object {
            "avatar": "https://bsky.public.url/img/avatar/plain/user(1)/cids(1)@jpeg",
            "did": "user(0)",
            "displayName": "ali",
            "handle": "alice.test",
            "labels": Array [
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-a",
              },
              Object {
                "cid": "cids(2)",
                "cts": "1970-01-01T00:00:00.000Z",
                "neg": false,
                "src": "user(0)",
                "uri": "record(1)",
                "val": "self-label-b",
              },
            ],
            "viewer": Object {
              "blockedBy": false,
              "muted": false,
            },
          },
          "cid": "cids(3)",
          "indexedAt": "1970-01-01T00:00:00.000Z",
          "labels": Array [],
          "likeCount": 3,
          "record": Object {
            "$type": "app.bsky.feed.post",
            "createdAt": "1970-01-01T00:00:00.000000Z",
            "text": "again",
          },
          "replyCount": 2,
          "repostCount": 1,
          "uri": "record(2)",
          "viewer": Object {},
        },
      },
    },
    Object {
      "post": Object {
        "author": Object {
          "avatar": "https://bsky.public.url/img/avatar/plain/user(1)/cids(1)@jpeg",
          "did": "user(0)",
          "displayName": "ali",
          "handle": "alice.test",
          "labels": Array [
            Object {
              "cid": "cids(2)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "user(0)",
              "uri": "record(1)",
              "val": "self-label-a",
            },
            Object {
              "cid": "cids(2)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "user(0)",
              "uri": "record(1)",
              "val": "self-label-b",
            },
          ],
          "viewer": Object {
            "blockedBy": false,
            "muted": false,
          },
        },
        "cid": "cids(7)",
        "embed": Object {
          "$type": "app.bsky.embed.record#view",
          "record": Object {
            "$type": "app.bsky.embed.record#viewRecord",
            "author": Object {
              "did": "user(5)",
              "handle": "dan.test",
              "labels": Array [
                Object {
                  "cts": "1970-01-01T00:00:00.000Z",
                  "neg": false,
                  "src": "did:example:labeler",
                  "uri": "user(5)",
                  "val": "repo-action-label",
                },
              ],
              "viewer": Object {
                "blockedBy": false,
                "following": "record(11)",
                "muted": false,
              },
            },
            "cid": "cids(8)",
            "embeds": Array [
              Object {
                "$type": "app.bsky.embed.record#view",
                "record": Object {
                  "$type": "app.bsky.embed.record#viewRecord",
                  "author": Object {
                    "did": "user(4)",
                    "handle": "carol.test",
                    "labels": Array [],
                    "viewer": Object {
                      "blockedBy": false,
                      "followedBy": "record(8)",
                      "following": "record(7)",
                      "muted": false,
                    },
                  },
                  "cid": "cids(9)",
                  "indexedAt": "1970-01-01T00:00:00.000Z",
                  "labels": Array [],
                  "uri": "record(12)",
                  "value": Object {
                    "$type": "app.bsky.feed.post",
                    "createdAt": "1970-01-01T00:00:00.000Z",
                    "embed": Object {
                      "$type": "app.bsky.embed.recordWithMedia",
                      "media": Object {
                        "$type": "app.bsky.embed.images",
                        "images": Array [
                          Object {
                            "alt": "tests/sample-img/key-landscape-small.jpg",
                            "image": Object {
                              "$type": "blob",
                              "mimeType": "image/jpeg",
                              "ref": Object {
                                "$link": "cids(5)",
                              },
                              "size": 4114,
                            },
                          },
                          Object {
                            "alt": "tests/sample-img/key-alt.jpg",
                            "image": Object {
                              "$type": "blob",
                              "mimeType": "image/jpeg",
                              "ref": Object {
                                "$link": "cids(10)",
                              },
                              "size": 12736,
                            },
                          },
                        ],
                      },
                      "record": Object {
                        "record": Object {
                          "cid": "cids(11)",
                          "uri": "record(13)",
                        },
                      },
                    },
                    "text": "hi im carol",
                  },
                },
              },
            ],
            "indexedAt": "1970-01-01T00:00:00.000Z",
            "labels": Array [],
            "uri": "record(10)",
            "value": Object {
              "$type": "app.bsky.feed.post",
              "createdAt": "1970-01-01T00:00:00.000Z",
              "embed": Object {
                "$type": "app.bsky.embed.record",
                "record": Object {
                  "cid": "cids(9)",
                  "uri": "record(12)",
                },
              },
              "facets": Array [
                Object {
                  "features": Array [
                    Object {
                      "$type": "app.bsky.richtext.facet#mention",
                      "did": "user(0)",
                    },
                  ],
                  "index": Object {
                    "byteEnd": 18,
                    "byteStart": 0,
                  },
                },
              ],
              "text": "@alice.bluesky.xyz is the best",
            },
          },
        },
        "indexedAt": "1970-01-01T00:00:00.000Z",
        "labels": Array [
          Object {
            "cid": "cids(7)",
            "cts": "1970-01-01T00:00:00.000Z",
            "neg": false,
            "src": "did:example:labeler",
            "uri": "record(9)",
            "val": "test-label",
          },
        ],
        "likeCount": 2,
        "record": Object {
          "$type": "app.bsky.feed.post",
          "createdAt": "1970-01-01T00:00:00.000Z",
          "embed": Object {
            "$type": "app.bsky.embed.record",
            "record": Object {
              "cid": "cids(8)",
              "uri": "record(10)",
            },
          },
          "text": "yoohoo label_me",
        },
        "replyCount": 0,
        "repostCount": 0,
        "uri": "record(9)",
        "viewer": Object {},
      },
    },
    Object {
      "post": Object {
        "author": Object {
          "avatar": "https://bsky.public.url/img/avatar/plain/user(3)/cids(1)@jpeg",
          "did": "user(2)",
          "displayName": "bobby",
          "handle": "bob.test",
          "labels": Array [],
          "viewer": Object {
            "blockedBy": false,
            "followedBy": "record(5)",
            "following": "record(4)",
            "muted": false,
          },
        },
        "cid": "cids(12)",
        "indexedAt": "1970-01-01T00:00:00.000Z",
        "labels": Array [],
        "likeCount": 0,
        "record": Object {
          "$type": "app.bsky.feed.post",
          "createdAt": "1970-01-01T00:00:00.000+00:00",
          "text": "bobby boy here",
        },
        "replyCount": 0,
        "repostCount": 0,
        "uri": "record(14)",
        "viewer": Object {},
      },
    },
    Object {
      "post": Object {
        "author": Object {
          "avatar": "https://bsky.public.url/img/avatar/plain/user(1)/cids(1)@jpeg",
          "did": "user(0)",
          "displayName": "ali",
          "handle": "alice.test",
          "labels": Array [
            Object {
              "cid": "cids(2)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "user(0)",
              "uri": "record(1)",
              "val": "self-label-a",
            },
            Object {
              "cid": "cids(2)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "user(0)",
              "uri": "record(1)",
              "val": "self-label-b",
            },
          ],
          "viewer": Object {
            "blockedBy": false,
            "muted": false,
          },
        },
        "cid": "cids(3)",
        "indexedAt": "1970-01-01T00:00:00.000Z",
        "labels": Array [],
        "likeCount": 3,
        "record": Object {
          "$type": "app.bsky.feed.post",
          "createdAt": "1970-01-01T00:00:00.000000Z",
          "text": "again",
        },
        "replyCount": 2,
        "repostCount": 1,
        "uri": "record(2)",
        "viewer": Object {},
      },
    },
    Object {
      "post": Object {
        "author": Object {
          "did": "user(4)",
          "handle": "carol.test",
          "labels": Array [],
          "viewer": Object {
            "blockedBy": false,
            "followedBy": "record(8)",
            "following": "record(7)",
            "muted": false,
          },
        },
        "cid": "cids(9)",
        "embed": Object {
          "$type": "app.bsky.embed.recordWithMedia#view",
          "media": Object {
            "$type": "app.bsky.embed.images#view",
            "images": Array [
              Object {
                "alt": "tests/sample-img/key-landscape-small.jpg",
                "fullsize": "https://bsky.public.url/img/feed_fullsize/plain/user(6)/cids(5)@jpeg",
                "thumb": "https://bsky.public.url/img/feed_thumbnail/plain/user(6)/cids(5)@jpeg",
              },
              Object {
                "alt": "tests/sample-img/key-alt.jpg",
                "fullsize": "https://bsky.public.url/img/feed_fullsize/plain/user(6)/cids(10)@jpeg",
                "thumb": "https://bsky.public.url/img/feed_thumbnail/plain/user(6)/cids(10)@jpeg",
              },
            ],
          },
          "record": Object {
            "record": Object {
              "$type": "app.bsky.embed.record#viewRecord",
              "author": Object {
                "avatar": "https://bsky.public.url/img/avatar/plain/user(3)/cids(1)@jpeg",
                "did": "user(2)",
                "displayName": "bobby",
                "handle": "bob.test",
                "labels": Array [],
                "viewer": Object {
                  "blockedBy": false,
                  "followedBy": "record(5)",
                  "following": "record(4)",
                  "muted": false,
                },
              },
              "cid": "cids(11)",
              "embeds": Array [],
              "indexedAt": "1970-01-01T00:00:00.000Z",
              "labels": Array [],
              "uri": "record(13)",
              "value": Object {
                "$type": "app.bsky.feed.post",
                "createdAt": "1970-01-01T00:00:00.000Z",
                "langs": Array [
                  "en-US",
                  "i-klingon",
                ],
                "text": "bob back at it again!",
              },
            },
          },
        },
        "indexedAt": "1970-01-01T00:00:00.000Z",
        "labels": Array [],
        "likeCount": 2,
        "record": Object {
          "$type": "app.bsky.feed.post",
          "createdAt": "1970-01-01T00:00:00.000Z",
          "embed": Object {
            "$type": "app.bsky.embed.recordWithMedia",
            "media": Object {
              "$type": "app.bsky.embed.images",
              "images": Array [
                Object {
                  "alt": "tests/sample-img/key-landscape-small.jpg",
                  "image": Object {
                    "$type": "blob",
                    "mimeType": "image/jpeg",
                    "ref": Object {
                      "$link": "cids(5)",
                    },
                    "size": 4114,
                  },
                },
                Object {
                  "alt": "tests/sample-img/key-alt.jpg",
                  "image": Object {
                    "$type": "blob",
                    "mimeType": "image/jpeg",
                    "ref": Object {
                      "$link": "cids(10)",
                    },
                    "size": 12736,
                  },
                },
              ],
            },
            "record": Object {
              "record": Object {
                "cid": "cids(11)",
                "uri": "record(13)",
              },
            },
          },
          "text": "hi im carol",
        },
        "replyCount": 0,
        "repostCount": 0,
        "uri": "record(12)",
        "viewer": Object {
          "like": "record(15)",
        },
      },
    },
    Object {
      "post": Object {
        "author": Object {
          "avatar": "https://bsky.public.url/img/avatar/plain/user(3)/cids(1)@jpeg",
          "did": "user(2)",
          "displayName": "bobby",
          "handle": "bob.test",
          "labels": Array [],
          "viewer": Object {
            "blockedBy": false,
            "followedBy": "record(5)",
            "following": "record(4)",
            "muted": false,
          },
        },
        "cid": "cids(11)",
        "indexedAt": "1970-01-01T00:00:00.000Z",
        "labels": Array [],
        "likeCount": 0,
        "record": Object {
          "$type": "app.bsky.feed.post",
          "createdAt": "1970-01-01T00:00:00.000Z",
          "langs": Array [
            "en-US",
            "i-klingon",
          ],
          "text": "bob back at it again!",
        },
        "replyCount": 0,
        "repostCount": 0,
        "uri": "record(13)",
        "viewer": Object {},
      },
    },
    Object {
      "post": Object {
        "author": Object {
          "avatar": "https://bsky.public.url/img/avatar/plain/user(1)/cids(1)@jpeg",
          "did": "user(0)",
          "displayName": "ali",
          "handle": "alice.test",
          "labels": Array [
            Object {
              "cid": "cids(2)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "user(0)",
              "uri": "record(1)",
              "val": "self-label-a",
            },
            Object {
              "cid": "cids(2)",
              "cts": "1970-01-01T00:00:00.000Z",
              "neg": false,
              "src": "user(0)",
              "uri": "record(1)",
              "val": "self-label-b",
            },
          ],
          "viewer": Object {
            "blockedBy": false,
            "muted": false,
          },
        },
        "cid": "cids(13)",
        "indexedAt": "1970-01-01T00:00:00.000Z",
        "labels": Array [
          Object {
            "cid": "cids(13)",
            "cts": "1970-01-01T00:00:00.000Z",
            "neg": false,
            "src": "user(0)",
            "uri": "record(16)",
            "val": "self-label",
          },
        ],
        "likeCount": 0,
        "record": Object {
          "$type": "app.bsky.feed.post",
          "createdAt": "1970-01-01T00:00:00.000Z",
          "labels": Object {
            "$type": "com.atproto.label.defs#selfLabels",
            "values": Array [
              Object {
                "val": "self-label",
              },
            ],
          },
          "text": "hey there",
        },
        "replyCount": 0,
        "repostCount": 0,
        "uri": "record(16)",
        "viewer": Object {},
      },
    },
  ],
}
`;
